### ISSUE TITLE
Unable to upload package

### ISSUE THREAD

**Issue by mogikanin** [NuGet/NuGetGallery#1839]
Hi,
i'm trying to upload package using "Upload" button at https://www.nuget.org/packages/upload, but i'm receiving "The package ID 'PackageName' is not available.". I checked that package name - it not used at nuget gallery.
Can you help me with this?

**Comment by MelbourneDeveloper** [comment]
Hi. I'm having a the same difficulty. Is there some way we can find out what the rules are for conflicting names?

I.e. Is it case sensitive? Does adding a dash or a dot stop conflicts? And if someone is using up the name but not actually publishing their product, shouldn't it be removed?

**Comment by skofman1** [comment]
Hi,
Sometimes packages are unlisted, in order to prevent their discovery be new users, but to still allow existing users to continue depending on them and not to brake their build. 
Adding a dash/dot is a valid solution to create a new package id.

**Comment by MelbourneDeveloper** [comment]
Yep. I tracked down the project and emailed the guy. I'll see what he says. 

If he is willing to transfer over the name, am I able to take it up?

**Comment by skofman1** [comment]
Sure. This is the correct course of action.

**Comment by MelbourneDeveloper** [comment]
I contacted the guy. He transferred ownership to me.  I am a happy chappy, and appreciate his kindness.

Anyway, this raises a really important point. In order for me to contact the guy, I had to send emails around to track down the guy who owned the NuGet ID. Could you guys please expose inactive NuGet packages over the web? Or, to protect the privacy of people who may not want their details exposed allow for some kind of messaging system so that we can message the owner and as for their permission to use the ID?

This was a perfect example of when people can be kind enough to give up an ID that may not be being used anymore. It would be really great if there were a mechanism through which we could contact the owner of a hidden NuGet package.

**Comment by scottbommarito** [comment]
I was under the impression that there already is a mechanism for contacting an owner of a hidden package. When a package is unlisted, you can still view the package by going to `https://www.nuget.org/packages/packageIdHere/`. Correct me if I'm wrong, but you should still be able to contact the package's owners by clicking on the "Contact Owners" link on that page. I definitely agree that it should be clearer to users who are trying to upload when there is a package with the same id that's unlisted and who to contact if they want ownership of the id, but ultimately we want to be careful about who sees the unlisted packages because a package is unlisted when its owner does not want additional people to see or use it.

**Comment by MelbourneDeveloper** [comment]
OK. Yes, you are correct. 

I guess then, all that would be nice would be a link to the unlisted package when you try to upload a package with the same ID. 

I tried to upload the package through the web front end, and it didn't tell me I could just look up the package when actually, I could. 

Also, it would be nice if any apps that handle publishing (maybe the packager?) could do the same thing. It would just save time and confusion.

**Comment by scottbommarito** [comment]
Yeah, when a user tries to upload a package with the same id of another package, it could be good to show a link to contact the owners of the existing package. @harikmenon any thoughts on this?

**Comment by FrankPfattheicher** [comment]
This problem may occur if you log in with your email address instead of user name or vice versa,
depending on how you are logged on creating the package initially.

**Comment by whoisj** [comment]
I am completely blocked by this issue.

I'm attempting to upload Microsoft.Alm.Authentication-signed.4.3.0.nupkg and I'm getting the nearly useless error message of 

> This package ID has been reserved. Please request access to upload to this reserved namespace from the owner of the reserved prefix, or re-upload the package with a different ID.

This is literally preventing from doing my job. Help please.

**Comment by skofman1** [comment]
@whoisj , Microsoft.* namespace is reserved for Microsoft account. 
All Microsoft packages are required to have "Microsoft" as a co-owner. Please reach out to us on support@nuget.org and we will provide further instructions. 
/cc @anangaur , @karann-msft

**Comment by whoisj** [comment]
@skofman1 I'll do so first thing Monday morning when I'm back in the office, on CorpNet, and have a non-phone to write emails with :wink:

**Comment by lkarun01** [comment]
I also have the same problem though i was able to successfully create the package successfully. But when i try to upload,  it Gives 'The package ID '<>' is not available.'
When i try to use CMD it gives 'Response status code does not indicate success: 403 (The specified API key is invalid, has expired, or does not have permission to access the specified package.).'

**Comment by skofman1** [comment]
@lkarun01 , what's the package id your are trying to upload? The error message indicates the id is already taken.

**Comment by dariogriffo** [comment]
Hi I'm having the same problem the name of the package trying to upload is
NetCore.DependencyInjection.Modules
code can be found [here](https://github.com/griffo-io/NetCore.DependencyInjection.Modules)

**Comment by skofman1** [comment]
@dariogriffo , the NetCore prefix is reserved to Microsoft. Please choose a different prefix for your package.

**Comment by baoduy** [comment]
I have no idea why the reserved ID is so troublesome. My package is HBD.AspNetCore.XXX is also not able to be uploaded. Why did Microsoft reserve my HBD organization prefix?

**Comment by anangaur** [comment]
@baoduy, its better to open a new issue so that it catches attention of the team trying to triage every day. To your question, there could be many reasons for it. The best way is to open a support ticket by sending an email to support @ nuget.org. I have opened a ticket copying you.

/cc: @clairernovotny
